surfactant/filetypeid/id_extension.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+import pathlib
+import re
+from typing import Optional
+
+import surfactant.plugin
+
+
+@surfactant.plugin.hookimpl
+def identify_file_type(filepath: str) -> Optional[str]:
+    # pylint: disable=too-many-return-statements
+    try:
+        with open(filepath, "r") as f:
+            head = f.readline().strip("\n")
+            suffix = pathlib.Path(filepath).suffix.lower()
+            # Check for script files
+            if re.match(r"#!.*bash", head) or suffix == ".sh":
```

Comment:
Two things here: for the `#!` check, `sh` should be recognized (probably) as `SHELL`, and same for a `.sh` extension (I've seen some file suffixes of `.bash` for bash scripts, but otherwise, the `.sh` doesn't indicate what interpreter should be used.